File: HW/HW6/ThreeSum.java
import java.util.*;

public class ThreeSum { 
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        Set<List<Integer>> set = new HashSet<>();
        int length = nums.length;

        for (int i = 0; i < length - 2; i++) {
            for (int j = i + 1; j < length - 1; j++) {
                for (int k = j + 1; k < length; k++) {
                    if ((nums[i] + nums[j] + nums[k]) == 0) {
                        List<Integer> temp = new ArrayList<>();
                        temp.add(nums[i]);
                        temp.add(nums[j]);
                        temp.add(nums[k]);
                        set.add(temp);
                    }
                }
            }
        }
        return new ArrayList<>(set);
    }

    public static void main(String[] args) {
        ThreeSum ts = new ThreeSum();
        int[] nums = {-5,0,5,10,-10,0};
        // int[] nums = {0,1,1};
        System.out.println(ts.threeSum(nums));
    }
}

File: HW/HW6/ThreeSum.py
from typing import List

class ThreeSum:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        setted = set()

        length = len(nums)
        for i in range(length-2):
            for j in range(i+1,length-1):
                for k in range(j+1,length):
                    if (nums[i]+nums[j]+nums[k]) == 0:
                        setted.add((nums[i],nums[j],nums[k]))
        
        if len(setted) == 0:
            return []
        else:
            return setted


three_sum = ThreeSum()
nums = [-5, 0, 5, 10, -10, 0] 
print(three_sum.threeSum(nums))
